PerimeterXModule/Internals/Helpers/PxCommonUtils.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Net;
+using System.Web;
+
+namespace PerimeterX
+{
+	class PxCommonUtils
+	{
+		/**
+		 * <summary>
+		 * Request helper, extracting the ip from the request according to sockerIpHeader or from 
+		 * the request socker when sockerIpHeader is absent
+		 * </summary>
+		 * <param name="context" >HttpContext</param>
+		 * <param name="pxConfig">PxConfiguration</param>
+		 * <returns>Ip from the request</returns>
+		 */
+		public static string GetRequestIP(HttpContext context, PxModuleConfigurationSection pxConfig)
+		{
+			// Get IP from custom header
+			string socketIpHeader = pxConfig.SocketIpHeader;
+			if (!string.IsNullOrEmpty(socketIpHeader))
+			{
+				var headerVal = context.Request.Headers[socketIpHeader];
+				if (headerVal != null)
+				{
+					var ips = headerVal.Split(new char[] { ',', ' ', '\t', '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries);
+					IPAddress firstIpAddress;
+					if (ips.Length > 0 && IPAddress.TryParse(ips[0], out firstIpAddress))
+					{
+						return ips[0];
```

Comment:
Can return `firstIpAddress` instead of `ips[0]` (although they are obviously the same), otherwise it feels a bit weird to declare this variable and populate it as part of `TryParse` without ever using it.

Reply:
thats how the api works, we also expect to have it as a string and firstIpAddress return it as object
thats why we use [0]